tests/x11/rstudio_server.pm

```diff
@@ -62,13 +62,19 @@ sub run() {
     # log out at last and close firefox
     assert_and_click("rstudio_server-sign-out");
     assert_screen("rstudio_server-login-username");
+
     send_key('alt-f4');
 
-    # optionally click away the close all tabs window
-    assert_screen([qw(rstudio_server-firefox_quit-and-close-tabs generic-desktop)]);
+    check_screen([qw(rstudio_server-firefox_quit-and-close-tabs generic-desktop)], timeout => 60);
     if (match_has_tag('rstudio_server-firefox_quit-and-close-tabs')) {
+        # optionally click away the close all tabs window => FF should close
         click_lastmatch();
         assert_screen('generic-desktop');
+    } elsif (!match_has_tag('generic-desktop')) {
+        # firefox is stuck and just greys out but does not display the "close
+        # all tabs" dialog => kill it
+        script_run('pkill -9 firefox');
```

Comment:
i dont know much about rstudio but this is what i am thinking. this sounds like a bug. So it would make sense to mark it with a softfail. 
Also, as it is now, 
- will check screen once 
- if doesnt find a match in 60s it will execute the `pkill`

what about something like
```
do {
        check_screen([qw(rstudio_server-firefox_quit-and-close-tabs rstudio_server-login-username generic-desktop)]);
        if (match_has_tag('rstudio_server-firefox_quit-and-close-tabs')) {
           click_lastmatch();
        }
        if (match_has_tag('rstudio_server-login-username')) {
              # if match means stalling
              #softfail or pkill or...
         }
    } while (not match_has_tag('generic-desktop'));
```

in additional i didnt find the `pkill` in the VR so i am not sure if this change actually works.

Reply:
I thought `check_screen` would check continuously in the same way that `assert_screen` does. Does it really work differently?

Reply:
they are the same in this matter. `check_screen` will try to match within a timeout period and it will continue. 
But based on the documentation If you take a look
, `check_screen` is preferrable  for `for optional matches` and `is recommended to use the lowest possible timeout`.

what i suggested was to run continually `check_screen` with default timeout , so you will continue as soon as it matches while waiting for some time.

Reply:
I really don't see the difference between running `check_screen([...], timeout => 60)` and `check_screen([...])` in a loop. What am I missing here?

Reply:
Only use `check_screen` if you want to check the return code. 

We also listen the preference for `assert_screen` in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style.

What the documentation means with "Use this for optional matches" is that you can use `check_screen` if there might be something on the screen and maybe not. But if there is something on the screen and it could be "foo" or "bar" use a multi-tag assert_screen (as in before). @b10n1k I am not sure if maybe you misunderstood what the documentation wants to say as well?

And, yes, a `check_screen 'foo', 60` can be equivalent to `for i in (1 .. 60) { last if check_screen 'foo', 60 }`. Both cases can waste 60s and should be avoided.